output/filewriter_test.go

```diff
@@ -62,3 +80,398 @@ func TestFileWriterConfig(t *testing.T) {
 		})
 	}
 }
+
+// testFileWriterCompareInOut sends numRecords records and wait for the given
+// duration between each send, to a FileWriter where PathString is set to the
+// concatenation of the path components.
+func testFileWriterCompareInOut(numRecords int, wait, rotate time.Duration, comps ...string) func(*testing.T) {
```

Comment:
we can probably add another check to verify that all the created files are exactly the same sent to the upload channel. It will make sure that no other files are created during the output execution.

Reply:
done

Reply:
done